src/main/java/org/broadinstitute/hellbender/engine/LocusWalker.java

```diff
@@ -0,0 +1,115 @@
+package org.broadinstitute.hellbender.engine;
+
+import htsjdk.samtools.SAMFileHeader;
+import htsjdk.samtools.SAMReadGroupRecord;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.downsampling.DownsamplingMethod;
+import org.broadinstitute.hellbender.utils.locusiterator.LocusIteratorByState;
+
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+/**
+ * A LocusWalker is a tool that processes reads that overlaps a single position in a reference at a time from
+ * one or multiple sources of reads, with optional contextual information from a reference and/or sets of
+ * variants/Features.
+ *
+ * LocusWalker authors must implement the apply() method to process each position, and may optionally implement
+ * onTraversalStart() and/or onTraversalDone().
+ *
+ * @author Daniel Gómez-Sánchez (magicDGS)
+ */
+public abstract class LocusWalker extends GATKTool {
+
+	/**
+	 * LocusWalkers requires read sources
+	 */
+	@Override
+	public boolean requiresReads() {
+		return true;
+	}
+
+	/**
+	 * Does this tool require deletions in the AlignmentContext? Tools that do should override to return true.
+	 *
+	 * @return {@code true} if this tool requires deletions, {@code false} otherwise
+	 */
+	public boolean includeDeletions() {
+		return true;
+	}
+
```

Comment:
Could you add an overridable method `makeReadFilter()` to this class, and have it by default return a `Wellformed` and `Mapped` filter? Eg.,

        public CountingReadFilter makeReadFilter(){
            return new CountingReadFilter("Wellformed", new WellformedReadFilter(getHeaderForReads()))
                    .and(new CountingReadFilter("Mapped", ReadFilterLibrary.MAPPED));
        }

To apply the filter in `traverse()` you'll probably need something like https://github.com/broadinstitute/gatk/blob/80f83ac52a61e3253537d2aa106fc8a84d48a671/src/main/java/org/broadinstitute/hellbender/utils/iterators/ReadFilteringIterator.java (from an unmerged branch), which I can extract from its branch into master so you can use it.